kyo-core/shared/src/main/scala/kyo/Hub.scala

```diff
@@ -3,30 +3,36 @@ package kyo
 import Hub.*
 import java.util.concurrent.CopyOnWriteArraySet
 
-/** A Hub is a multi-producer, multi-consumer channel that broadcasts messages to all listeners.
+/** A multi-producer, multi-consumer primitive that broadcasts messages to multiple listeners.
+  *
+  * Hub provides fan-out functionality by delivering each published message to all active listeners. Unlike a regular Channel which delivers
+  * messages to a single consumer, Hub enables multiple consumers to receive and process the same messages.
+  *
+  * Message flow and buffering:
+  *   - Publishers send messages to the Hub's main buffer
+  *   - A dedicated fiber distributes messages from the Hub buffer to each listener's individual buffer
+  *   - Each listener consumes messages from its own buffer at its own pace
+  *   - When any listener's buffer becomes full and the Hub's buffer is also full, backpressure is applied to publishers, affecting message
+  *     delivery to all listeners
   *
   * @tparam A
-  *   the type of elements in the Hub
+  *   The type of messages that can be published through this Hub
   */
-class Hub[A] private[kyo] (
+final class Hub[A] private[kyo] (
     ch: Channel[A],
     fiber: Fiber[Closed, Unit],
-    listeners: CopyOnWriteArraySet[Channel[A]]
+    listeners: CopyOnWriteArraySet[Listener[A]]
 )(using initFrame: Frame):
 
-    /** Returns the current size of the Hub.
-      *
-      * @return
-      *   the number of elements currently in the Hub
-      */
-    def size(using Frame): Int < (IO & Abort[Closed]) = ch.size
```

Comment:
I've removed `size` because it has poor semantics. Since a hub has both the main buffer and the listener buffers, this method can be confusing given that it can indicate it's the total number of messages in the hub. We'll likely need to expose more information but it's probably better to do that via stats/metrics.